### ISSUE TITLE
[ Dartify ] Using type specific to firestore

### ISSUE THREAD

**Issue by tperraut** [FirebaseExtended/firebase-dart#363]
The helper dartify is using firestore specific types like GeoPoint forcing to import the firebase-firestore.js dependencie in index.html without actually needing it.

Thank you for your work.

**Comment by kevmoo** [comment]
Good tip!

**Comment by kevmoo** [comment]
What happens if you **don't** include `firebase-firestore.js` in your app?

**Comment by tperraut** [comment]
It is spamming with Geopoint missing warnings.

Le mer. 16 déc. 2020 à 22:25, Kevin Moore <notifications@github.com> a
écrit :

**Comment by kevmoo** [comment]
In the console? Hrm...

I'm not quite sure what to do here. I could just ignore the cast I
guess...hrm...

**Comment by tperraut** [comment]
Yes in the console, pretty problematic in our app as it is really intensive on position changes. I think the dartify helper should not cast to GeoPoint if it is a Firebase realtime object and not a Firestore GeoPoint. I am not an expert of Firestore but for what I understood, a Geopoint is more than a simple object with latitude and longitude as people seems to have issues when creating an object with latitude and longitude and not marking it as a GeoPoint. (cf. https://github.com/FirebaseExtended/firebase-dart/issues/288#issuecomment-705250000)

**Comment by kevmoo** [comment]
@tperraut – PR pending.

If you're adventurous, please pull it in as a dependency_override and check to see if it addresses things!
